block/block_sync.go

```diff
@@ -221,12 +221,11 @@ func (bSyncService *BlockSyncService) Start() error {
 
 // Stop is a part of Service interface.
 func (bSyncService *BlockSyncService) Stop() error {
-	err := bSyncService.blockStore.Stop(bSyncService.ctx)
-	err = errors.Join(
-		err,
+	err := errors.Join(
 		bSyncService.p2pServer.Stop(bSyncService.ctx),
 		bSyncService.ex.Stop(bSyncService.ctx),
 		bSyncService.sub.Stop(bSyncService.ctx),
+		bSyncService.blockStore.Stop(bSyncService.ctx),
```

Comment:
We should capture the learnings here in a comment around the importances of closing the store last so that in the future if we add services, we close those services prior to closing the store if they use the store.